### PR TITLE
Fix: fork segment/amplitude 5.2.2 to force cookies to have a 1 year expiration date

### PR DESCRIPTION
**Pull request by rsylvian**
Ticket: [DD-261](https://circleci.atlassian.net/browse/DD-261)

# Description
- Fork Segment and Amplitude to be able to set a 1-year expiration date for cookies (instead of 10)

# Reasons
For GDRP reasons we want to limit amplitude cookies to be 1 year long instead of 10. This is not an option amplitude gives with `v5.2.2` so we had to fork it. They do offer such an option in `v6.0.0` but it introduces breaking changes (with segment) that we are not ready to handle given the urgency of that work.

### CODE REVIEW

**Review by Nivl — APPROVED** [review]
Looks good!

File: jekyll/_includes/segment.html [hunk 1]
```diff
@@ -1,5 +1,8 @@
 <script type="text/javascript">
-  !function(){var analytics=window.analytics=window.analytics||[];if(!analytics.initialize)if(analytics.invoked)window.console&&console.error&&console.error("Segment snippet included twice.");else{analytics.invoked=!0;analytics.methods=["trackSubmit","trackClick","trackLink","trackForm","pageview","identify","reset","group","track","ready","alias","debug","page","once","off","on"];analytics.factory=function(t){return function(){var e=Array.prototype.slice.call(arguments);e.unshift(t);analytics.push(e);return analytics}};for(var t=0;t<analytics.methods.length;t++){var e=analytics.methods[t];analytics[e]=analytics.factory(e)}analytics.load=function(t){var e=document.createElement("script");e.type="text/javascript";e.async=!0;e.src=("https:"===document.location.protocol?"https://":"http://")+"cdn.segment.com/analytics.js/v1/"+t+"/analytics.min.js";var n=document.getElementsByTagName("script")[0];n.parentNode.insertBefore(e,n)};analytics.SNIPPET_VERSION="4.0.0";
+  !function(){
+    var isProduction = window.location.origin === 'https://circleci.com';
```
Comment (ccgeller) [inline jekyll/_includes/segment.html:3]: Do browsers automatically remove www. ?

File: .circleci/config.yml [hunk 2]
```diff
@@ -300,8 +300,8 @@ jobs:
             set -exu
             mkdir -p /tmp/workspace/js
             mkdir -p /tmp/workspace/css
-            mv /tmp/workspace/js/* jekyll/assets/js/
-            mv /tmp/workspace/css/* jekyll/assets/css/
+            mv -fn /tmp/workspace/js/* jekyll/assets/js/
+            mv -fn /tmp/workspace/css/* jekyll/assets/css/
```
Comment (rsylvian) [inline .circleci/config.yml:304]: `-fn`: `f` forces and `n` doesn't overwrite. CI was giving us trouble without that

File: jekyll/_includes/segment.html [hunk 3]
```diff
@@ -1,5 +1,8 @@
...
+    var segmentFileUrl = isProduction ? 'https://circleci.com/docs/assets/js/segment/analytics-AbgkrgN4cbRhAVEwlzMkHbwvrXnxHh35.min.js' : 'https://cdn.segment.com/analytics.js/v1/mAJ9W2SwLHgmJtFkpaXWCbwEeNk9D8CZ/analytics.min.js';
+    var analytics=window.analytics=window.analytics||[];if(!analytics.initialize)if(analytics.invoked)window.console&&console.error&&console.error("Segment snippet included twice.");else{analytics.invoked=!0;analytics.methods=["trackSubmit","trackClick","trackLink","trackForm","pageview","identify","reset","group","track","ready","alias","debug","page","once","off","on"];analytics.factory=function(t){return function(){var e=Array.prototype.slice.call(arguments);e.unshift(t);analytics.push(e);return analytics}};for(var t=0;t<analytics.methods.length;t++){var e=analytics.methods[t];analytics[e]=analytics.factory(e)}analytics.load=function(t){var e=document.createElement("script");e.type="text/javascript";e.async=!0;e.src=segmentFileUrl;var n=document.getElementsByTagName("script")[0];n.parentNode.insertBefore(e,n)};analytics.SNIPPET_VERSION="4.0.0";
```
Comment (ccgeller) [inline jekyll/_includes/segment.html:5]:
```suggestion
    var analytics = window.analytics=window.analytics||[];if(!analytics.initialize)if(analytics.invoked)window.console&&console.error&&console.error("Segment snippet included twice.");else{analytics.invoked=!0;analytics.methods=["trackSubmit","trackClick","trackLink","trackForm","pageview","identify","reset","group","track","ready","alias","debug","page","once","off","on"];analytics.factory=function(t){return function(){var e=Array.prototype.slice.call(arguments);e.unshift(t);analytics.push(e);return analytics}};for(var t=0;t<analytics.methods.length;t++){var e=analytics.methods[t];analytics[e]=analytics.factory(e)}analytics.load=function(t){var e=document.createElement("script");e.type="text/javascript";e.async=!0;e.src=segmentFileUrl;var n=document.getElementsByTagName("script")[0];n.parentNode.insertBefore(e,n)};analytics.SNIPPET_VERSION="4.0.0";
```

File: jekyll/_includes/segment.html (hunk 1 above)
Comment (nramabad) [reply jekyll/_includes/segment.html:3]: do we want to inline this with the rest of the script?

**Review by nramabad — APPROVED** [review]
LGTM! left one question.

File: jekyll/_includes/segment.html (hunk 1 above)
Comment (rsylvian) [reply jekyll/_includes/segment.html:3]: we do not have `www` on circleci.com:

Screen Shot 2021-12-06 at 12 48 02 PM

**Review by tshen815 — APPROVED** [review]

File: jekyll/_includes/segment.html (hunk 1 above)
Comment (rsylvian) [reply jekyll/_includes/segment.html:3]: @nramabad I don't think so. line `5` was the original snippet and line 3-4 is the additional "patch" we introduced. I'll add a comment tho

**Review by ccgeller — APPROVED** [review]
LGTM 🙏🌺🤯

**Pull request merged**